components/purchase_modal/purchase_modal.tsx

```diff
@@ -115,10 +115,13 @@ function findProductInDictionary(products: Dictionary<Product> | undefined, prod
     return currentProduct;
 }
 
-function getSelectedProduct(products: Dictionary<Product> | undefined, productId?: string | null) {
+function getSelectedProduct(products: Dictionary<Product> | undefined, productId?: string | null, isFreeTrial?: boolean | null) {
     const currentProduct = findProductInDictionary(products, productId);
+    if (isFreeTrial) {
+        return currentProduct;
+    }
     let nextSku = CloudProducts.PROFESSIONAL;
-    if (currentProduct?.sku === CloudProducts.PROFESSIONAL) {
+    if (currentProduct?.sku !== CloudProducts.STARTER) {
```

Comment:
Given the condition above for free trial, do we want to leave this condition as before? Not sure what the logic is here

Reply:
yes, you are right, with the isFreeTrial condition, the previous validation makes more sense. Updating it now.